ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/copy-coordinates/copy-coordinates.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+import { hot } from 'react-hot-loader'
+import Dropdown from '../dropdown'
+import MenuAction from '../menu-action'
+import NavigationBehavior from '../navigation-behavior'
+import { ContextType } from '../../../react-component/presentation/dropdown'
+
+const Clipboard = require('clipboard')
+const announcement = require('component/announcement')
+
+type Props = {
+  clickLat: string
+  clickLon: string
+  clickDms: string
+  clickMgrs: string
+  clickUtmUps: string
+  closeParent: () => void
+}
+
+const Text = styled.div`
+  line-height: 1.2rem;
+`
+
+const Description = styled.div`
+  opacity: ${props => props.theme.minimumOpacity};
+`
+
+const generateClipboardHandler = (
+  text: string,
+  context: ContextType,
+  closeParent: () => void
+) => {
+  return (e: React.MouseEvent) => {
+    const clipboardInstance = new Clipboard(e.target, {
+      text: () => {
+        return text
+      },
+    })
+    clipboardInstance.on('success', function(e: any) {
```

Comment:
:question: What is `e`?

Reply:
`e` is an event with the `text` field added, which contains the copied coordinate string on the clipboard.